avaframe/ana5Utils/distanceTimeAnalysis.py

```diff
@@ -0,0 +1,653 @@
+
+        Parameters
+        -------------
+        flowF: numpy nd array
+            flow parameter result field
+        rangeMasked: masked numpy array
+            masked array of radar range
+        threshold: float
+            used to create mask for getting line of sight range,
+            data below threshold not taken into account
+
+        Returns
+        --------
+        losRange: float
+            minimum distance in line of sight to avalanche front
+            defined by the flowF
+
+    """
+
+    # fetch mask where result field smaller than threshold
+    maskAva = np.ma.masked_where(flowF < threshold, flowF)
+    bmaskAva = np.ma.getmask(maskAva)
+    # fetch mask of radar field of view
+    maskPhi = np.ma.getmask(rangeMasked)
+
+    # merge masks of result field smaller threshold and outside radar field of view
+    maskFull = ~np.logical_and(~maskPhi, ~bmaskAva)
+
+    # first get full data of the masked radar range array
+    r = np.ma.getdata(rangeMasked)
+    # mask this with full mask (result below threshold and outside field of view)
+    radarRange = np.ma.array(r, mask=maskFull)
+
+    # line of sight min of masked array
+    losRange = radarRange.min()
+
+    return losRange
+
+
+def extractMeanValuesAtRangeGates(cfgRangeTime, flowF, mtiInfo):
+    """ extract average values of flow parameter result at certain distance intervals (range gates)
+        add info to be used for colorcoding range-time diagram
+
+        Parameters
+        -----------
+        cfgRangeTime: configparser object
+            configuration settings for creating range-time diagram
+        flowF: numpy array
+            flow parameter result field
+        mtiInfo: dict
+            info here used: rArray, bmaskRadar rangeMasked, rangeGates, mti, timeList
+
+        Returns
+        --------
+        mtiInfo: dict
+            updated mtiInfo dict with info on mti values
+    """
+
+    # load parameters from configuration file and mtiInfo
+    rgWidth = cfgRangeTime['GENERAL'].getfloat('rgWidth')
+    # if threshold = 0.0 mean over all cells - if threshold > 0.0 mean only over all cells where there
+    # is avalanche flow
+    threshold = cfgRangeTime['GENERAL'].getfloat('thresholdResult')
+    rangeGates = mtiInfo['rangeGates']
+    rArray = mtiInfo['rArray']
+    bmaskRadar = mtiInfo['bmaskRadar']
+    rangeMasked = mtiInfo['rangeMasked']
+    mti = mtiInfo['mti']
+    mtiNew = np.zeros((len(rangeGates), 1))
+
+    # get mask of results below threshold
+    maskAva = np.ma.masked_where(flowF < threshold, flowF)
+    bmaskAva = np.ma.getmask(maskAva)
+    # and join to a mask with "visible part of avalanche" - mask where not field of view and not
+    # ava result above threshold
+    bmaskAvaRadar = ~np.logical_and(~bmaskRadar, ~bmaskAva)
+    # mask range with this mask to obtain 'visible part of avalanche' ranges
+    rmaskAva_radar= np.ma.array(np.ma.getdata(rangeMasked), mask=bmaskAvaRadar) # masked ranges
+
```

Comment:
this is exactly the same as minRangeSimulation(). Could we keep the rmaskAva_radar saved so that we can reuse it?

Reply:
made a function that is now called